src/cpp/rtps/transport/shared_mem/RobustExclusiveLock.hpp

```diff
@@ -0,0 +1,220 @@
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _FASTDDS_ROBUST_EXCLUSIVE_LOCK_H_
+#define _FASTDDS_ROBUST_EXCLUSIVE_LOCK_H_
+
+#ifdef  _MSC_VER
+#include <io.h>
+#else
+#include <sys/file.h>
+#endif
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <boostconfig.hpp>
+#include <boost/interprocess/detail/shared_dir_helpers.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+
+/**
+ * This class implement an interprocess named resource that holds a mutual exclusion lock until
+ * destroyed, or until the creator process dies
+ */
+class RobustExclusiveLock
+{
+public:
+
+    /**
+     * Create the interprocess lock.
+     * @throw std::exception if a RobustExclusiveLock with name is alive.
+     */
+    RobustExclusiveLock(
+        const std::string& name,
+        bool* was_lock_created)
+    {
+        auto file_path = get_file_path(name);
+
+        fd_ = open_and_lock_file(file_path, was_lock_created);
+
+        name_ = name;
+    }
+
+    /**
+     * Create the interprocess lock.
+     * @throw std::exception if a RobustExclusiveLock with name is alive.
+     */
+    RobustExclusiveLock(
+            const std::string& name)
+    {
+        bool was_lock_created;
+
+        auto file_path = get_file_path(name);
+
+        fd_ = open_and_lock_file(file_path, &was_lock_created);
+
+        name_ = name;
+    }
+
+    /**
+     *  Unlock the interprocess lock.
+     */
+    ~RobustExclusiveLock()
+    {
+        unlock_and_close();
+    }
+
+private:
+
+    std::string name_;
+    int fd_;
+
+#if !defined(BOOST_INTERPROCESS_POSIX_SHARED_MEMORY_OBJECTS)
+
+    std::string get_file_path(  
+            const std::string& file_name)
+    {
+        std::string shmfile;
+        boost::interprocess::ipcdetail::shared_filepath(file_name.c_str(), shmfile);
+        return shmfile;
+    }
+
+#else
+
+    std::string get_file_path(
+            const std::string& filename)
+    {
+        std::string filepath;
+        #if defined(BOOST_INTERPROCESS_FILESYSTEM_BASED_POSIX_SHARED_MEMORY)
+        const bool add_leading_slash = false;
+        #elif defined(BOOST_INTERPROCESS_RUNTIME_FILESYSTEM_BASED_POSIX_SHARED_MEMORY)
+        const bool add_leading_slash = !boost::interprocess::shared_memory_object_detail::use_filesystem_based_posix();
+        #else
+        const bool add_leading_slash = true;
+        #endif
+        if (add_leading_slash)
+        {
+            boost::interprocess::ipcdetail::add_leading_slash(filename.c_str(), filepath);
+        }
+        else
+        {
+            boost::interprocess::ipcdetail::shared_filepath(filename.c_str(), filepath);
```

Comment:
this function is really a copy/paste from boost::interprocess internals. I only added the "/dev/shm". I feel better not making changes in the function because it is the same code used by `boost::interprocess::shared_memory_object::remove`